e2e/bazel/bazel.go

```diff
@@ -0,0 +1,180 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"regexp"
+
+	blaze_query "github.com/bazelbuild/bazel-watcher/third_party/bazel/master/src/main/protobuf"
+	"github.com/golang/protobuf/proto"
+)
+
+func fuzzyEqual(pattern []interface{}, compare []string) bool {
+	if len(pattern) != len(compare) {
+		return false
+	}
+
+	for k, _ := range pattern {
+		switch v := pattern[k].(type) {
+		case *regexp.Regexp:
+			if !v.MatchString(compare[k]) {
+				return false
+			}
+			break
+		case string:
+			if v != compare[k] {
+				return false
+			}
+			break
+		default:
+			fmt.Errorf("I have no idea what a %T is... please fix.", v)
+			panic(v)
+		}
+	}
+
+	return true
+}
+
+func runWithScript(args []string) {
+	// Since all ru ncommands start with "run" as arg0 and it breaks flag
+	// parsing, strip args[1]. If you need it, imagine there is a "run" at the
+	// beginning.
+	args = args[1:]
+
+	fs := flag.NewFlagSet("bazel_run_flags", flag.ExitOnError)
+	scriptPath := fs.String("script_path", "default", "")
+	err := fs.Parse(args)
+	if err != nil {
+		panic(err)
+	}
+
+	sp, err := os.OpenFile(*scriptPath, os.O_CREATE|os.O_WRONLY, 0755)
+	if err != nil {
+		panic(err)
+	}
+
+	err = sp.Chmod(0755)
+	if err != nil {
+		panic(err)
+	}
+
+	fmt.Fprintf(sp, `#! /usr/bin/env bash
```

Comment:
Move this into a helper function with a descriptive name. Its a bit hard to transition from reading beautiful go code to a *insert your own adjective to describe bash syntax* bash script